feat: fix for `privatednszone` name in `web/static-site/`

ChrisSidebotham opened the pull request:
## Description

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Closes #123
-->

Closes #5985 

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
| [![avm.res.web.static-site](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml/badge.svg?branch=statsite-pdns&event=workflow_dispatch)](https://github.com/ChrisSidebotham/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml)|

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [x] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

ChrisSidebotham committed:
adding deployment files

ChrisSidebotham committed:
ficing rp ref

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Adding workflow file

ChrisSidebotham committed:
Merge branch 'Azure:main' into main

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/ChrisSidebotham/bicep-registry-modules

ChrisSidebotham committed:
fixing path

ChrisSidebotham committed:
Fixing Workflow name

ChrisSidebotham committed:
adding workflow

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Adding Workflow file

ChrisSidebotham committed:
Adding workflow file

ChrisSidebotham committed:
Merge branch 'anchor' of https://github.com/ChrisSidebotham/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Adding workflow file

ChrisSidebotham committed:
adding workflow

ChrisSidebotham committed:
Fixing workflow name

ChrisSidebotham committed:
update workflow

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
updating Test Case s location to fix Pipeline

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
adding workflow

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Adding workflow

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Update forwardTo property to null if empty

ChrisSidebotham committed:
Update templateHash values in Service Bus Namespace and Topic

ChrisSidebotham committed:
Merge branch 'main' into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'Azure:main' into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'Azure:main' into main

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
#2578

ChrisSidebotham committed:
typo

ChrisSidebotham committed:
typo

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'anchor' of https://github.com/ChrisSidebotham/bicep-registry-modules into anchor

ChrisSidebotham committed:
Adding logic to skip for non resource modules

ChrisSidebotham committed:
chore: Remove trailing whitespace in main.json file

ChrisSidebotham committed:
Merge branch 'anchor' of https://github.com/ChrisSidebotham/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Fixing minimumTTL

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
Merge branch 'main' of https://github.com/Azure/bicep-registry-modules into anchor

ChrisSidebotham committed:
CHORE: Clean up code structure and remove redundancies

- Refactored several functions for improved readability.
- Removed unused variables and imports throughout the codebase.
- Updated comments to better reflect the current logic.
- Ensured consistent formatting across all files.

AlexanderSehr commented on avm/res/web/static-site/main.bicep:
```diff
@@ -274,10 +274,10 @@ resource staticSite_roleAssignments 'Microsoft.Authorization/roleAssignments@202
   name: '${uniqueString(deployment().name, location)}-staticSite-PrivateDnsZone'
   params: {
-    name: 'privatelink.${staticSite.properties.defaultHostname}.azurestaticapps.net'
+    name: 'privatelink.${split(staticSite.properties.defaultHostname, '.')[1]}.azurestaticapps.net'
```
Hey @ChrisSidebotham assuming the `defaultHostName` has a value of like `"happy-sea-15afae3e.azurestaticwebsites.net"`, would you not need to refer to the `[0]` item of the split array 
`[`
`  happy-sea-15afae3e,`
`  azurestaticwebsites,`
`  net`
`]`

?

ChrisSidebotham committed:
Merge remote-tracking branch 'upstream/main' into statsite-pdns

ChrisSidebotham committed:
Fix pr update

ChrisSidebotham committed:
CHANGE: avm/res/web/static-site - Remove placeholder for version 0.9.5

- Removed the changelog entry for version 0.9.5, which included a fix for the host name duplication bug.

ChrisSidebotham committed:
Merge branch 'main' into statsite-pdns

ChrisSidebotham committed:
Merge branch 'main' into statsite-pdns

ChrisSidebotham committed:
Merge branch 'main' into statsite-pdns
